src/main/java/at/hannibal2/skyhanni/features/garden/farming/GardenCustomKeybinds.kt

```diff
@@ -67,43 +79,106 @@ object GardenCustomKeybinds {
         lastDuplicateKeybindsWarnTime = SimpleTimeMark.now()
     }
 
+    @HandleEvent
+    fun onGardenToolChange(event: GardenToolChangeEvent) {
+        lastToolSwitch = SimpleTimeMark.now()
+        cropInHand = event.crop
+        event.crop?.let { lastCrop = it }
+        currentLayout = layouts[cropLayoutSelection[cropInHand]]
+    }
+
+//     TODO: remove the need for this workaround, as GardenAPI should call GardenToolChangeEvent on island change
+//     @HandleEvent
+//     fun onIslandChange(event: IslandChangeEvent) {
+//         if (event.newIsland == IslandType.GARDEN) {
+//             DelayedRun.runDelayed(2.seconds) {
+//                 cropInHand = GardenAPI.cropInHand
+//                 currentLayout = layouts[cropLayoutSelection[cropInHand]]
+//             }
+//         }
+//     }
+
+    @JvmStatic
+    fun getAllKeybindingsFromLayout(layout: KeyBindLayout) = listOf(
+        layout.attack, layout.useItem, layout.left, layout.right,
+        layout.forward, layout.back, layout.jump, layout.sneak
+    )
```

Comment:
An extension function works better here. (Or make it a member function in java, with an ArrayList/Array)
```suggestion
    @JvmStatic
    fun KeyBindLayout.getAllKeybindings() = listOf(
        this.attack, this.useItem, this.left, this.right,
        this.forward, this.back, this.jump, this.sneak
    )
```

Reply:
done